apps/webservice/src/app/[workspaceSlug]/systems/[systemSlug]/_components/EditSystemDialog.tsx

```diff
@@ -0,0 +1,126 @@
+"use client";
+
+import React, { useState } from "react";
+
+import * as schema from "@ctrlplane/db/schema";
+import { Button } from "@ctrlplane/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@ctrlplane/ui/dialog";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormRootError,
+  useForm,
+} from "@ctrlplane/ui/form";
+import { Input } from "@ctrlplane/ui/input";
+import { Textarea } from "@ctrlplane/ui/textarea";
+
+import { api } from "~/trpc/react";
+
+export const EditSystemDialog: React.FC<{
+  system: schema.System;
+  children: React.ReactNode;
+  onSubmit?: () => void;
+}> = ({ system, children, onSubmit }) => {
+  const [open, setOpen] = useState(false);
+
+  const form = useForm({
+    schema: schema.updateSystem,
+    defaultValues: {
+      name: system.name,
+      slug: system.slug,
+      description: system.description,
+    },
+  });
+
+  const updateSystem = api.system.update.useMutation();
+
+  const onFormSubmit = form.handleSubmit((data) => {
+    updateSystem.mutate(
+      { id: system.id, data },
+      {
+        onSuccess: () => {
+          onSubmit?.();
+          setOpen(false);
+        },
+        onError: () => {
+          form.setError("root", {
```

Comment:
what if the error is something else?

Reply:
If you try and edit it and you haven't gotten the form correct, you will get this:

<img width="768" alt="Screenshot 2024-09-27 at 9 59 45 AM" src="https://github.com/user-attachments/assets/381d965d-ebde-46ef-b117-f1343be40b43">

If it is another reason, I don't think we have a good pattern yet to show other messages.